keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
 
 ```
-message ProvisionerGrantBucketAccessRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
-    
-    map<string,string> bucket_context = 2;  
+    ProvisionerGetInfo
+    |------------------------------------------|       |---------------------------------------|
+    | grpc ProvisionerGetInfoRequest{}         | ===>  | ProvisionerGetInfoResponse{           |
+    |------------------------------------------|       |   "name": "s3.amazonaws.com"          |
+                                                       | }                                     |
+                                                       |---------------------------------------|
+```
 
-    string principal = 3;
-    
-    // This field is REQUIRED
-    string access_policy = 4;
-}
+#### ProvisionerCreateBucket
+
+This gRPC call creates a bucket in the OSP, and returns information about the new bucket. This api must be idempotent. The input to this call is the name of the bucket and an opaque parameters field.
+
+The returned `bucketID` should be a unique identifier for the bucket in the OSP. This value could be the name of the bucket too. This value will be used by COSI to make all subsequent calls related to this bucket.
 
-message ProvisionerGrantBucketAccessResponse {
-    // This is the account that is being provided access. This will
-    // be required later to revoke access. 
-    string principal = 1;
-    
-    string credentials_file_contents = 2;
-    
-    string credentials_file_path = 3;
-} 
 ```
+    ProvisionerCreateBucket
+    |------------------------------------------|       |-----------------------------------------------|
+    | grpc ProvisionerCreateBucketRequest{     | ===>  | ProvisionerCreateBucketResponse{              |
+    |     "name": "br-$uuid",                  |       |   "bucketID": "br-$uuid",                     |
+    |     "parameters": {                      |       |   "bucketInfo": {                             |
+    |        "key": "value"                    |       |      "s3": {                                  |
+    |     }                                    |       |        "bucketName": "br-$uuid",              |
+    | }                                        |       |        "region": "us-west1",                  |
+    | -----------------------------------------|       |        "endpoint": "s3.amazonaws.com"         |
+                                                       |      }                                        |
+                                                       |   }                                           |
+                                                       | }                                             |
+                                                       |-----------------------------------------------|
+```
+
+#### ProvisionerGrantBucketAccess
```

Comment:
Should we define any default permissions if nothing is specified? Or require the driver to error?

Reply:
the BucketAccessClass is  a required field. We shouldn't need a default ideally

Reply:
I mean what happens if `BucketAccessClass.parameters` is empty?

Reply:
The default is chosen by the driver or it can error out if it so chooses.

Reply:
We should be more prescriptive about what the expected behavior should be from a driver and pick one.